packages/core/src/lib/connection_manager.ts

```diff
@@ -361,15 +349,13 @@ export class ConnectionManager
     );
   }
 
-  public async attemptDial(peerId: PeerId): Promise<void> {
-    if (!(await this.shouldDialPeer(peerId))) return;
-
-    if (this.currentActiveParallelDialCount >= this.options.maxParallelDials) {
-      this.pendingPeerDialQueue.push(peerId);
-      return;
+  private startBrowserNetworkStatusListener(): void {
+    try {
+      window.addEventListener("online", this.onEventHandlers["online"]);
+      window.addEventListener("offline", this.onEventHandlers["offline"]);
```

Comment:
I'm concerned about relying solely on window events to detect when there is an internet connection. Is it possible for the event listener to miss an online event? This could cause the app to be stuck in offline state when there is actually a connection present.

It might be worth manually checking the boolean value of `navigator.onLine` as a fallback when trying to take an action that depends on `isConnectedToNetwork`

Reply:
though I don't believe in `online`/`offline` event being out of sync with `navigator.online` as it would violate browser guarantees - I am removing duplication of the state on our side, should be safer indeed!